fud/fud/stages/interpreter.py

```diff
@@ -3,8 +3,7 @@
 from pathlib import Path
 import simplejson as sjson
 import numpy as np
-from fud.stages.verilator.numeric_types import FixedPoint, Bitnum
-from fud.errors import InvalidNumericType
+from calyx.numeric_types import FixedPoint, Bitnum, InvalidNumericType
```

Comment:
It could be described as slightly weird, although not overwhelmingly weird, that fud depends on `calyx-py` now. Like, if you think of the `calyx-py` library as being for generating code, it's not clear why fud would need that. But actually, `calyx-py` currently is sort of sprawling and covers lots of things, including all our individual code generators (it is not just a code generation library). Therefore this doesn't seem like much messier than it was before, so I think it's fine!

Reply:
Hm, I think we should eventually externalize all these stages from the core logic in `fud` so that these false dependencies can be eliminated. The `fud` API is rich enough that we don't actually need to have these stages sitting in the definition of `fud` itself anymore.

Reply:
Is the idea to have all stages be external then? Or rather to get rid of the notion of non-external stages?

Reply:
Yeah, I'd say barring any API problems, there should be no internal stages whatsoever.